### PR TITLE
string format http event validation and removal

### LINKED ISSUE #3402: Deprecate and remove simple string-based http event representation
# This is a Feature Proposal

## Description

The current string representation of the `http` event (`http: GET myPath`) is super limiting and nearly always useless (because most of the time you need `CORS` support anyway).

The string representation should be removed. Here are the necessary steps:

- [ ] Update the docs
- [ ] Log a message when someone is still using the string representation
- [ ] Add a note about the change in the list of breaking changes
- [ ] Remove support after ~x updates to the framework

Refs #2718

/cc @eahefnawy @DavidWells @brianneisler

### PR DESCRIPTION
**Pull request by patil16nit**
## What did you implement:

Closes #3402

## How did you implement it:
http event string validation removal for upcoming version

## How can we verify it:
```yml
# serverless.yml
functions:
  index:
    handler: handler.hello
    events:
      - http: GET myPath
```
This will give validation error.
   

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** YES

### CODE REVIEW

**Comment by pmuens** [comment]
Super nice! Thanks for getting this started @patil16nit 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
+# 1.10.0
```
Comment (vladgolubev) [inline CHANGELOG.md]: Version `1.11.0` is out today, so... needs a small update 😄

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+## Breaking Changes
+### Removed string representation of http event
+We have removed string representation of of http event(http: GET myPath). It should be specified in object format.
```
Comment (vladgolubev) [inline CHANGELOG.md]: Typo: `of of`

**Comment by pmuens** [comment]
Thanks for the review @vladgolubev 👍 

I just updated the PR so that it's GTM. However we need to find a strategy here so that users know that they should update (if they use the string event definition).

/cc @eahefnawy @brianneisler

**Comment by vladgolubev** [comment]
@pmuens what about https://github.com/facebook/jscodeshift ?

**Comment by pmuens** [comment]
@vladgolubev thanks for the response.

That's a good idea. We thought about something like warning the user on the CLI if he uses the old syntax and provide a simple update guide, but having this done automatically sounds more appealing.

**Comment by pmuens** [comment]
@patil16nit thanks for taking the time and effort to get this started 👍

Really appreciate it! We've just discussed it yesterday. The problem with this change is that it's a breaking change and we've committed to try to avoid breaking changes whenever we can.

The new plan is to enable CORS automatically when using the simple `http` string event definition (the lack of CORS support was one of the main motivations to remove the sting representation in the first place).

More information about this can be found here: https://github.com/serverless/serverless/issues/3486

Thanks again for being so kind to get this started! 💯

**Pull request closed without merging**